src/Microsoft.ML.Data/Transforms/KeyToVector.cs

```diff
@@ -246,14 +246,24 @@ public Mapper(KeyToVectorMappingTransformer parent, DataViewSchema inputSchema)
                 _parent = parent;
                 _infos = CreateInfos(inputSchema);
                 _types = new VectorDataViewType[_parent.ColumnPairs.Length];
-                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                try
                 {
-                    int valueCount = _infos[i].TypeSrc.GetValueCount();
-                    int keyCount = _infos[i].TypeSrc.GetItemType().GetKeyCountAsInt32(Host);
-                    if (_parent._columns[i].OutputCountVector || valueCount == 1)
-                        _types[i] = new VectorDataViewType(NumberDataViewType.Single, keyCount);
-                    else
-                        _types[i] = new VectorDataViewType(NumberDataViewType.Single, valueCount, keyCount);
+                    for (int i = 0; i < _parent.ColumnPairs.Length; i++)
```

Comment:
Can you please add a corresponding test for this? And in the test, please mention the bug that prompted this change.

Reply:
Can we put user's dataset into our tests? Since this exception is triggered by specific dataset and  I can not reproduce the same error using a self-designed dataset

Reply:
If possible, please try to avoid using user data set. From the description it sound like it was because OneHotHashEncoding was applied to the date column with a lot of unique values. Is it not possible to reproduce that? 

---
In reply to: [439126305](https://github.com/dotnet/machinelearning/pull/5232#discussion_r439126305) [](ancestors = 439126305)